jira/client.py

```diff
@@ -2631,9 +2631,28 @@ def delete_user(self, username):
     def deactivate_user(self, username):
         """Disable/deactivate the user."""
         if self.deploymentType == 'Cloud':
-            url = self._options['server'] + '/admin/rest/um/1/user/deactivate?username=' + username
-            self._options['headers']['Content-Type'] = 'application/json'
-            userInfo = {}
+            # Disabling users now needs cookie auth in the Cloud - see https://jira.atlassian.com/browse/ID-6230
+            if 'authCookie' not in vars(self):
+                user = self.session()
+                if user.raw is None:
+                    auth_method = (
+                        oauth or basic_auth or jwt or kerberos or "anonymous"
+                    )
+                    raise JIRAError("Can not log in with %s" % str(auth_method))
+                self.authCookie = '%s=%s' % (user.raw['session']['name'], user.raw['session']['value'])
+            url = self._options['server'] + '/admin/rest/um/1/user/deactivate?username=%s' % (username)
+            # We can't use our existing session here - this endpoint is fragile and objects to extra headers
+            try:
+                r = requests.post(url, headers={'Cookie': self.authCookie, 'Content-Type': 'application/json'}, data={})
+                if r.status_code == 200:
+                    return True
+                else:
+                    logging.warning(
+                        'Got response from deactivating %s: %s' % (username, r.status_code))
+                    return r.status_code
+            except urllib2.HTTPError as e:
+                logging.error(
+                        "Error Deactivating %s: %s" % (username, e))
```

Comment:
I am not sure we want to catch exceptions and hide them. I think that we should raise a JiraError instead, or at least return False. Otherwise consumers of this function would not be able to know if call succeeded or not.

Same applies for the code below.

Also I wonder if we can avoid duplicating so much code between cloud mode and self-hosted one.